core/storage/topic/gae_models.py

```diff
@@ -112,7 +112,11 @@ def get_by_name(cls, topic_name):
         Returns:
             TopicModel. The summary model of the question.
         """
-        return TopicModel.query().filter(cls.name == topic_name).fetch()
+        topic = TopicModel.query().filter(cls.name == topic_name).fetch()
+        if not topic:
```

Comment:
if topic is None...

actually, no, I think your varname is misleading. Why are you referring to topic[0]?

Be careful with your varnames. Make sure they're conceptually correct. Bad varnames cause confusion and muddy the code.

Reply:
Here the returned query is a list, but that'll always contain only one element. So what data is queried is named "topic". And the check is for checking if the list is empty. And if it's not, then we just need to return the first element.

Maybe I update the docstring for this?

Reply:
You can't call a list with one element a singular variable. A list represents a plural.

Perhaps use get() instead of fetch()?

Reply:
Yes, Done. Thanks! I don't know why I wasn't using get :)